### PR TITLE
Bring back Qubole

### PR DESCRIPTION
**Pull request by wild-endeavor**

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -9,8 +9,9 @@ ignored = ["k8s.io/spark-on-k8s-operator",
 
 [[constraint]]
   name = "github.com/lyft/flyteidl"
-  source = "https://github.com/lyft/flyteidl"
-  version = "^0.1.x"
+  source = "git@github.com:lyft/flyteidl"
+  # version = "^0.1.x"
+  revision = "603420a216676e0d0f5e0f6bc0b203ee8eea79ca"
```
Comment (wild-endeavor) [inline Gopkg.toml]: I'm pinning this for now.  This is just https://github.com/lyft/flyteidl/pull/3

File: go/tasks/v1/resourcemanager/redis_resource_manager.go [hunk 2]
```diff
@@ -9,12 +9,12 @@ import (
...
 	"github.com/prometheus/client_golang/prometheus"
 
-	"github.com/lyft/flyteplugins/go/tasks/v1/qubole/config"
+	"github.com/lyft/flyteplugins/go/tasks/v1/qubole_collection/config"
 )
 
 // This is the key that will point to the Redis Set.
 // https://redis.io/commands#set
-const RedisSetKey = "qubole"
+const RedisSetKey = "qubole_collection"
```
Comment (wild-endeavor) [inline go/tasks/v1/resourcemanager/redis_resource_manager.go]: not correct, came in with goland refactor, but we're getting rid of this file.

File: go/tasks/v1/qubole_single/secrets_manager.go [hunk 3]
```diff
@@ -10,7 +10,7 @@ import (
 
 	"github.com/lyft/flytestdlib/logger"
 
-	"github.com/lyft/flyteplugins/go/tasks/v1/qubole/config"
+	"github.com/lyft/flyteplugins/go/tasks/v1/qubole_collection/config"
```
Comment (wild-endeavor) [inline go/tasks/v1/qubole_single/secrets_manager.go]: we're going to get rid of all the secrets managers.

File: go/tasks/hive/execution_state.go [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+	case PhaseQueryFailed:
+		return "PhaseQueryFailed"
+	}
+	return "Bad Qubole execution phase"
+}
+
+type ExecutionState struct {
+	Phase ExecutionPhase
+
+	// This ID is the cache key and so will need to be unique across all objects in the cache (it will probably be
```
Comment (kumare3) [inline go/tasks/hive/execution_state.go]: we should not need to store this, this is derivable right?

File: go/tasks/hive/execution_state.go [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+	// This will also be used as the allocation token for now.
+	Id string `json:"id"`
+
+	// This will store the command ID from Qubole
+	CommandId string `json:"command_id,omitempty"`
+
+	// This number keeps track of the number of failures within the sync function. Without this, what happens in
+	// the sync function is entirely opaque. Note that this field is completely orthogonal to Flyte system/node/task
+	// level retries, just errors from hitting the Qubole API, inside the sync loop
+	SyncQuboleApiFailures int `json:"sync_qubole_api_failures,omitempty"`
```
Comment (kumare3) [inline go/tasks/hive/execution_state.go]: if we are using json, I prefer to have small names please. `sync-failures` `create-failures`?
Also omitempty works with pointers, you do not need it here as 0 is valid

**Comment by kumare3** [comment]
there are a few qubole collection thingys

**Pull request merged**